# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1258638017 -3600
#      Thu Nov 19 14:40:17 2009 +0100
# Node ID 3e8b4c80ae6323d91b81b12892a93c1ed1665263
# Parent  55219e65c7cda226c8f3bf05a635c097df778541
use Array::get_size in Array::clear

diff --git a/liboctave/Array.cc b/liboctave/Array.cc
--- a/liboctave/Array.cc
+++ b/liboctave/Array.cc
@@ -97,17 +97,17 @@ Array<T>::clear (void)
 
 template <class T>
 void
 Array<T>::clear (const dim_vector& dv)
 {
   if (--rep->count <= 0)
     delete rep;
 
-  rep = new ArrayRep (dv.numel ());
+  rep = new ArrayRep (get_size (dv));
   slice_data = rep->data;
   slice_len = rep->len;
 
   dimensions = dv;
 }
 
 template <class T>
 Array<T>
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2009-11-19  Jaroslav Hajek  <highegg@gmail.com>
+
+	* Array.cc (Array::clear (const dim_vector&)): Use get_size.
+
 2009-11-19  Jaroslav Hajek  <highegg@gmail.com>
 
 	* lo-mappers.cc (do_erfinv): New static function.
 	(erfinv (double), erfinv (float)): New mappers.
 	* lo-mappers.h: Declare them.
 
 2009-11-18  David Grundberg  <davidg@cs.umu.se>
 
